vivarium/core/engine.py

```diff
@@ -850,22 +860,48 @@ def update(
             self,
             interval: float
     ) -> None:
-        """Run each process for the given interval and update the states.
         """
-
-        time = 0.0
-        emit_time = self.emit_step
+        Run each process for the given interval and update the states.
+        """
         clock_start = clock.time()
+        self.run_for(interval=interval, force_complete=True)
+        self.check_complete()
+        clock_finish = clock.time() - clock_start
```

Comment:
To me, `start` and `finish` usually mean absolute times, so `finish - start = delta`. Here, `clock_finish` is more of what I would think of as the delta